pymaker/__init__.py

```diff
@@ -584,6 +592,10 @@ def transact(self, **kwargs) -> Optional[Receipt]:
                 # Check if any transaction sent so far has been mined (has a receipt).
                 # If it has, we return either the receipt (if if was successful) or `None`.
                 for attempt in range(1, 11):
+                    if self.replaced:
```

Comment:
Checking this bool should be atomic; I don't see a need to obtain `transaction_lock`, because `nonce` must already be assigned before it gets here.